src/harness/testcases/WINNF_FT_S_PCR_testcase.py

```diff
@@ -92,34 +60,79 @@ def isPpaWithinServiceArea(pal_records, ppa_zone_geometry):
   return ppa_zone_shapely_geometry.buffer(-1e-6).within(pal_service_area)
 
 
-class PpaCreationTestcase(sas_testcase.SasTestCase):
-  """Implementation of PCR tests to verify the area of the non-overlapping
-  difference between the maximum PPA boundary created by SAS UUT shall be no more than
-  10% of the area of the maximum PPA boundary created by the Reference Model.
+def assertRegConditionalsForPpaRefModel(registration_requests,
+                                        conditional_registration_datas):
+  """Check the REG Conditionals for PPA creation model and raises an exception.
+
+  Performs the assert to check installationParam present in
+  registrationRequests or conditional registration data and raises an exception.
+  PpaCreationModel requires the input registrationRequests to have
+  'installationParam'. But this parameter is removed for devices where
+  conditionals are pre-loaded. Adding the 'installationParam' into
+  registrationRequests by taking the corresponding values from
+  conditionalRegistrationData.
+
+  Args:
+    registration_requests: A list of individual CBSD registration
+      requests (each of which is itself a dictionary).
+    conditional_registration_datas: A list of individual CBSD registration
+      data that need to be preloaded into SAS (each of which is a dictionary).
+      data that need to be preloaded into SAS (each of which is a dictionary).
+      the fccId and cbsdSerialNumber fields are required, other fields are
+      optional but required for ppa reference model.
+
+  Raises:
+    Exception: If the installationParam object and required
+      fields is not found in conditionalRegistrationData and registrationRequests
+      for category B then raises an exception.
+
   """
-  def setUp(self):
-    self._sas, self._sas_admin = sas.GetTestingSas()
-    self._sas_admin.Reset()
+  for device in registration_requests:
+    if 'installationParam' not in device:
+      install_param_assigned = False
+      for conditional_params in conditional_registration_datas:
+        # Check if FCC_ID+Serial_Number present in registrationRequest
+        # and conditional_params match and add the 'installationParam'.
+        if (conditional_params['fccId'] == device['fccId'] and
+                conditional_params['cbsdSerialNumber'] == device['cbsdSerialNumber']):
+          device.update({'installationParam': conditional_params['installationParam']})
+          install_param_assigned = True
+          # If the cbsdCategory is not present in registration request then
+          # assign it to the cbsdCategory in conditional_params.
+          if 'cbsdCategory' not in device:
+            device.update({'cbsdCategory': conditional_params.get('cbsdCategory',
+                                                                  'B')})
```

Comment:
There is no default value for cbsdCategory.

Reply:
changed to break out as suggested.